cli/azd/cmd/package.go

```diff
@@ -142,15 +156,7 @@ func (pa *packageAction) Run(ctx context.Context) (*actions.ActionResult, error)
 			continue
 		}
 
-		packageTask := pa.serviceManager.Package(ctx, svc, nil)
-		go func() {
-			for packageProgress := range packageTask.Progress() {
-				progressMessage := fmt.Sprintf("Packaging service %s (%s)", svc.Name, packageProgress.Message)
-				pa.console.ShowSpinner(ctx, progressMessage, input.Step)
-			}
-		}()
-
-		packageResult, err := packageTask.Await()
+		packageResult, err := pa.serviceManager.Package(ctx, svc, nil)
```

Comment:
Results of service level funcs become much easier to work with and you don't need to be concerned with manually draining progress channels.